terraform/environments/data-platform/iam.tf

```
@@ -136,40 +203,21 @@ data "aws_iam_policy_document" "iam_policy_document_for_get_glue_metadata_lambda
     actions   = ["logs:CreateLogGroup", "logs:CreateLogStream", "logs:PutLogEvents"]
     resources = ["arn:aws:logs:${local.region}:${local.account_id}:log-group:/aws/lambda/*"]
   }
-  statement {
-    sid    = "s3LogAccess"
-    effect = "Allow"
-    actions = [
-      "s3:GetObject",
-      "s3:PutObject",
-    ]
-    resources = [
-      "${module.s3-bucket.bucket.arn}/logs/*"
-    ]
-  }
 }
 
 data "aws_iam_policy_document" "iam_policy_document_for_presigned_url_lambda" {
+  source_policy_documents = [data.aws_iam_policy_document.log_to_bucket.json, data.aws_iam_policy_document.read_metadata.json]
+
   statement {
     sid     = "GetPutDataObject"
     effect  = "Allow"
     actions = ["s3:GetObject", "s3:PutObject"]
     resources = [
-      "${module.s3-bucket.bucket.arn}/raw_data/*",
-      "${module.s3-bucket.bucket.arn}/logs/*"
+      "${module.data_s3_bucket.bucket.arn}/raw/*",
+      "${module.logs_s3_bucket.bucket.arn}/logs/*"
```

```suggestion
      "${module.logs_s3_bucket.bucket.arn}/logs/*",
      "${module.data_s3_bucket.bucket.arn}/raw",
      "${module.logs_s3_bucket.bucket.arn}/logs",
```
paired with the above suggestion for bucket access